### PR TITLE
reduce test setup in azframework tests

### PR DESCRIPTION
**Pull request by Kadino**
Proposed changes were found while profiling tests, and reduce running AZ::AzFramework.Tests.main by 57% from 92752ms to 39867ms. Other tests in this module appear more tightly integrated with creating an actual application context.

Reduces sleeps when testing FileIO, though a more invasive fix would be to mock the system time responses that FileIO is depending on from the OS, or to mock more of the actual I/O. This would also prevent a rare edge case failure where actual system time is modified during test execution (could be user-initiated, though more likely automated in the case of DST, leap seconds, and internal clock drift)

Application start/stop was not required to test compression.

Verified these changes locally in both profile and debug.

Signed-off-by: Sweeney <sweeneys@amazon.com>

### CODE REVIEW

File: Code/Framework/AzFramework/Tests/ArchiveCompressionTests.cpp [hunk 1]
```diff
@@ -35,29 +35,6 @@ namespace UnitTest
...
-            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
-        }
-
-        void TearDown() override
-        {
-            m_application->Stop();
-        }
-
     private:
         AZStd::unique_ptr<AzFramework::Application> m_application;
```
Comment (amzn-mike) [inline Code/Framework/AzFramework/Tests/ArchiveCompressionTests.cpp:39]: Should this be removed as well?

Comment (Kadino) [reply Code/Framework/AzFramework/Tests/ArchiveCompressionTests.cpp:39]: The application object still needs to be constructed for this test to work, but not started. Though I'm sure all of these tests can be rewritten without depending directly on AzFramework::Application

Comment (amzn-mike) [reply Code/Framework/AzFramework/Tests/ArchiveCompressionTests.cpp:39]: Maybe add a comment about that, otherwise it looks unused.

**Review by amzn-mike — APPROVED** [review]

File: Code/Framework/AzFramework/Tests/ArchiveCompressionTests.cpp (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Code/Framework/AzFramework/Tests/ArchiveCompressionTests.cpp:39]: The [AzFramework Application constructor](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Application/Application.cpp#L116-L134) initializes the AZ::IO::Archive(CryPak) and the AZ::IO::FileIOBase to setup the FileIO Aliases, so it is not needed to invoke Start and Stop.
FYI the base class Component Application constructor initializes the Settings Registry so that is available as well.

Now the important part of the changes to note is that because we don't invoke `Application::Start` anymore, the [UserSettingsComponent](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Application/Application.cpp#L338) is not created, therefore we don't have to worry about it trying to write to the `<project-path>/user/UserSettings.xml` when the `Application::Stop` is invoked.

TLDR: This change should be fine.

**Review by lumberyard-employee-dm — APPROVED** [review]
